File: components/credentials/src/main/java/org/cloudfoundry/credhub/views/SshView.java
package org.cloudfoundry.credhub.views;

import org.cloudfoundry.credhub.credential.SshCredentialValue;
import org.cloudfoundry.credhub.domain.SshCredentialVersion;

@SuppressWarnings("unused")
public class SshView extends CredentialView {

  SshView() {
    super(); /* Jackson */
  }

  SshView(final SshCredentialVersion sshCredential) {
    super(
      sshCredential.getVersionCreatedAt(),
      sshCredential.getUuid(),
      sshCredential.getName(),
      sshCredential.getCredentialType(),
      new SshCredentialValue(sshCredential.getPublicKey(), sshCredential.getPrivateKey(),
        sshCredential.getFingerprint())
    );
  }
}

File: components/credentials/src/main/kotlin/org/cloudfoundry/credhub/views/SshView.kt
package org.cloudfoundry.credhub.views

import org.cloudfoundry.credhub.credential.SshCredentialValue
import org.cloudfoundry.credhub.domain.SshCredentialVersion

class SshView : CredentialView {
    internal constructor() : super() /* Jackson */ {}
    internal constructor(sshCredential: SshCredentialVersion) : super(
        sshCredential.versionCreatedAt,
        sshCredential.uuid,
        sshCredential.name,
        sshCredential.getCredentialType(),
        SshCredentialValue(sshCredential.publicKey, sshCredential.privateKey,
            sshCredential.fingerprint)
    ) {
    }
}
